pub-matic-htb.js

```diff
@@ -141,20 +249,25 @@ function PubMaticHtb(configs) {
          */
 
         /* ---------------------- PUT CODE HERE ------------------------------------ */
-        var queryObj = {};
-        var callbackId = System.generateUniqueId();
-
-        /* Change this to your bidder endpoint.*/
-        var baseUrl = Browser.getProtocol() + '//someAdapterEndpoint.com/bid';
-
-        /* ---------------- Craft bid request using the above returnParcels --------- */
-
+        var payload = {},
+        callbackId = System.generateUniqueId(),
+        baseUrl = Browser.getProtocol() + '//hbopenbid.pubmatic.com/translator?source=index-client';
+        payload = {
+            id: '' + new Date().getTime(), // str | mandatory
+            at: 1, // int | mandatory
+            cur: ['USD'], // [str] | opt
+            imp: __populateImprObject(returnParcels), // obj | mandatory - pending
+            site: __populateSiteObject(returnParcels[0].pubId), //// obj | opt
+            device: __populateDeviceInfo(returnParcels[0]), // obj | mandatory
+            user: __populateUserInfo(returnParcels[0]), // obj | opt
+            ext: __populateExtObject(returnParcels[0]), // not required?? - to be checked
+            secure: Browser.getProtocol() === "https:" ? 1 : 0
+        }
```

Comment:
returnParcels are objects that contain different slots for which demand needs to be requested. A returnParcel object looks something like this:

```
{
    "partnerId": "PUBM",
    "htSlot": {
      "__type__": "HeaderTagSlot"
    },
    "ref": "googletag.Slot",   
    "xSlotRef": {   
      "adUnitName": "pubmatic_test@728x90",
      "sizes": [ [728, 90], [970, 250] ]
    },
    "xSlotName": "1",
    "requestId": "_fwyvecpA" 
}
```
This object does not have reference of the `pubId` property. As a result,  `returnParcels[0].pubId` is `undefined`.

Reply:
pubId is part of the partner config, and is required in the request payload. Hence it is read from the config, and added to the return parcels in the generateReturnParcels method in generateRequestObject.spec.js file. 
![screenshot](https://user-images.githubusercontent.com/29033921/38798422-3ac74a68-417f-11e8-87dd-7845b01b851c.png)

the resultant returnParcels should have a pubId property in it and would look like this:  
{
	"pubId": "301",
	"partnerId": "PubMaticHtb",
	"partnerStatsId": "PUBM",
	"htSlot": {
		"id": "htSlotID-1"
	},
	"ref": "",
	"xSlotName": "xSlot1",
	"xSlotRef": {
		"adUnitName": "/15671365/DMDemo",
		"sizes": [
			[300, 250]
		],
		"bid_id": "_9ruakzhod"
	},
	"requestId": "_x21ewezlv",
	"bid_id": "_9ruakzhod",
	"lat": "40.712775",
	"lon": "-74.005973",
	"yob": "1982",
	"gender": "M",
	"kadfloor": "1.75"
}

Please let us know, if you encounter a case where pubId is coming as undefined.

Reply:
made the required changes.

Reply:
done